toolchain/check/interface_support.cpp

```diff
@@ -0,0 +1,86 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/interface_support.h"
+
+#include "toolchain/check/inst.h"
+#include "toolchain/check/merge.h"
+#include "toolchain/check/name_lookup.h"
+#include "toolchain/sem_ir/entity_with_params_base.h"
+
+namespace Carbon::Check {
+
+auto GetExistingDeclOrDiagnoseMismatch(
+    Context& context, Parse::NodeId node_id, const NameComponent& name,
```

Comment:
```suggestion
    Context& context, SemIR::LocId loc_id, const NameComponent& name,
```

Looks like the node_id may only be used for a loc_id, maybe switch on the call boundary? That'll also make it look a little more idiomatic versus other functions.

Reply:
Done